test/boost/sstable_test.cc

```diff
@@ -63,26 +63,17 @@ SEASTAR_TEST_CASE(composite_index) {
     return test_using_working_sst(composite_schema(), "test/resource/sstables/composite");
 }
 
-template<typename Func>
-inline future<>
-test_using_reusable_sst(schema_ptr s, sstring dir, sstables::generation_type::int_t gen, Func&& func) {
-    return test_env::do_with_async([s = std::move(s), dir = std::move(dir), gen, func = std::move(func)] (test_env& env) {
-        auto sst = env.reusable_sst(std::move(s), std::move(dir), generation_from_value(gen)).get();
-        func(env, std::move(sst));
-    });
-}
-
-template<typename T, typename Func>
-inline future<T>
-test_using_reusable_sst_returning(schema_ptr s, sstring dir, sstables::generation_type::int_t gen, Func&& func) {
-    return test_env::do_with_async_returning<T>([s = std::move(s), dir = std::move(dir), gen, func = std::move(func)] (test_env& env) {
+inline auto
+test_using_reusable_sst(schema_ptr s, sstring dir, sstables::generation_type::int_t gen, std::invocable<test_env&, sstable_ptr> auto&& func) {
```

Comment:
NIT: Isn't the wording "de-templatize" misleading? What happens in this commit is actually constraining the template parameter with a concept, but it's still a template

Reply:
You are correct. This syntax combines the _abbreviated function template syntax_ (which necessitates the `decltype()` later on) with a _type constraint_, resulting in _type-constrained auto_. But that doesn't change `test_using_reusable_sst_returning()` from being a function template to a non-template function. I'll remove the de-templatize language from the commit message of v2. Thanks!

Reply:
... I will not change the branch name though (so technically speaking, the branch name will be a misnomer starting with v2, but I can't change the branch name without abandoning this PR).

Reply:
resolved in v2